backend/src/submission/submission.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from "@nestjs/common";
+
+import { PrismaService } from "src/core/prisma.service";
+
+@Injectable()
+export class SubmissionService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async getSubmissionsById(id: string) {
+    const roomSession = await this.prisma.roomSession.findUnique({
+      where: { id: id },
```

Comment:
Let's also do an authorisation check here:
```suggestion
      where: { id: id, users: { some: { id: userId } } },
```
We can return a `404` if the room is not found.

Reply:
This code is getting an error. I am guessing it is because `users` is not a unique field so it cannot be inside `where` clause.

From prisma docs: Wraps all unique fields of a model so that individual records can be selected.
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#options-2

Reply:
Yeah we will need to do a `FindFirst` instead I believe.

Reply:
Ah I see. Thanks!